jenkins/bisect.jpl

```diff
@@ -424,13 +424,110 @@ def checkAbort(passed, message) {
     return passed
 }
 
-node("bisection") {
+def bisection(kci_core, kdir, checks) {
+    def check = null
+
+    stage("Init") {
+        timeout(time: 30, unit: 'MINUTES') {
+            parallel(
+                kci_core: { cloneKciCore(kci_core) },
+                kdir: { cloneLinux(kdir) },
+            )
+        }
+    }
+
+    stage("Check pass") {
+        check = runCheck(kdir, kci_core, params.GOOD_COMMIT, 'pass', 0)
+    }
+    if (!checkAbort(check, "Good revision check failed"))
+        return
+
+    stage("Check fail") {
+        check = runCheck(kdir, kci_core, params.BAD_COMMIT, 'fail', 2)
+    }
+    if (!checkAbort(check, "Bad revision check failed"))
+        return
+
+    stage("Start") {
+        timeout(time: 5, unit: 'MINUTES') {
+            check = bisectStart(kdir)
+        }
+    }
+    if (!checkAbort(check,
+                    "Failed to start bisection, commits range may be invalid."))
+        return
+
+    def previous = params.GOOD_COMMIT
+    def current = getSHA(kdir)
+    def iteration = 1
+
+    while (previous != current) {
+        def tag = createTag(kdir, iteration)
+        def status = null
+
+        echo "Iteration #${iteration}: ${tag}"
+
+        lock("${env.NODE_NAME}-build-lock") {
+            stage("Build ${iteration}") {
+                timeout(time: 60, unit: 'MINUTES') {
+                    try {
+                        buildKernel(kdir, kci_core)
+                        status = 0
+                    } catch (error) {
+                        status = 1
+                    }
+                }
+            }
+        }
+
+        if (status == 0) {
+            def describe = gitDescribe(kdir)
+
+            stage("Test ${iteration}") {
+                timeout(time: 120, unit: 'MINUTES') {
+                    status = runTest(kci_core, describe)
+                }
+            }
+        }
+
+        removeTag(kdir, tag)
+
+        stage("Next") {
+            timeout(time: 5, unit: 'MINUTES') {
+                bisectNext(kdir, status)
+            }
+        }
+
+        previous = current
+        current = getSHA(kdir)
+        iteration += 1
+    }
+
+    stage("Verify") {
+        check = runCheck(kdir, kci_core, 'refs/bisect/bad', 'verify', 2,3)
+        checks['verify'] = check ? 'PASS' : 'FAIL'
+    }
+    if (!checkAbort(check, "Result check failed"))
+        return
+
+    stage("Revert") {
+        dir(kdir) {
+            sh(script: "git revert refs/bisect/bad")
+        }
+        check = runCheck(kdir, kci_core, 'HEAD', 'revert', 0, 3)
+        checks['revert'] = check ? 'PASS' : 'FAIL'
+    }
+    if (!check)
+        echo "Warning: revert check failed"
+}
+
+node("docker && bisection") {
     /* Global pipeline constants */
     env._BUILD_JSON = "build-data.json"
 
+    def docker_image = "${params.DOCKER_BASE}${params.COMPILER}_${params.ARCH}"
```

Comment:
I'm not sure I fully understand how the docker images are found/created, but this doesn't look quite right with the new hierarchy of Dockerfiles.